android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -214,10 +214,15 @@ constructor(
             facts.getFact(relatedResourceKey).value as List<Resource>
           else emptyList()
 
-      return value.filter {
-        resource.logicalId ==
-          fhirPathDataExtractor.extractValue(it, referenceFhirPathExpression).extractLogicalIdUuid()
-      }
+      return value
+        .asSequence()
```

Comment:
Using a sequence here is not useful when we eventually collect the results in a list.